src/backend/api/reservations.js

```diff
@@ -0,0 +1,90 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+// GET return all reservations
+
+router.get("/", async (request, response) => {
+    try {
+      const reservations = await knex("Reservation").select("*");
+      response.json(reservations);
+    } catch (error) {
+      response.status(500).json({ error: "Error retrieving reservations" });
+    }
+  });
+
+
+// POST Adds a new reservation to the database ??
+
+router.post("/", async (request, response) => {
+    const newReservation = request.body;
```

Comment:
In situations like this, it is safer to clone the request body. Notice that because the body is an object, we are not cloning it here, and merely storing its reference in a new variable. See here: https://www.freecodecamp.org/news/primitive-vs-reference-data-types-in-javascript/

The reason is that a function should never modify its parameters, otherwise the code could become unpredictable (imagine this in a larger scenario where you have multiple function calls, etc).

So, instead, I would recommend doing this:
`const newReservation = {...request.body};`